### PR TITLE
fixed authorization and authentifications

### PR DESCRIPTION
**Pull request by Boburm1rzo**

### CODE REVIEW

File: Web/Ecommerce/Ecommerce.Infrastructure/Persistence/Migrations/20240919170434_identities.cs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+            migrationBuilder.AddColumn<string>(
+                name: "Password",
+                table: "Customer",
+                type: "nvarchar(max)",
+                nullable: false,
+                defaultValue: "");
+        }
+    }
+}
```
Comment (jamshidbek7375) [inline Web/Ecommerce/Ecommerce.Infrastructure/Persistence/Migrations/20240919170434_identities.cs:203]: ok great

File: Web/Ecommerce/Ecommerce.Infrastructure/Persistence/Migrations/EcommerceDbContextModelSnapshot.cs [hunk 2]
```diff
@@ -72,8 +77,34 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
+                    b.Property<bool>("PhoneNumberConfirmed")
+                        .HasColumnType("bit");
+
+                    b.Property<string>("SecurityStamp")
+                        .HasColumnType("nvarchar(max)");
+
+                    b.Property<bool>("TwoFactorEnabled")
+                        .HasColumnType("bit");
+
+                    b.Property<string>("UserName")
```
Comment (jamshidbek7375) [inline Web/Ecommerce/Ecommerce.Infrastructure/Persistence/Migrations/EcommerceDbContextModelSnapshot.cs:107]: ok great

File: Web/Ecommerce/Ecommerce.Infrastructure/Persistence/Migrations/EcommerceDbContextModelSnapshot.cs [hunk 3]
```diff
@@ -305,6 +335,204 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
+                    b.Property<string>("Name")
+                        .HasColumnType("nvarchar(450)");
+
+                    b.Property<string>("Value")
+                        .HasColumnType("nvarchar(max)");
+
+                    b.HasKey("UserId", "LoginProvider", "Name");
+
+                    b.ToTable("AspNetUserTokens", (string)null);
+                });
```
Comment (jamshidbek7375) [inline Web/Ecommerce/Ecommerce.Infrastructure/Persistence/Migrations/EcommerceDbContextModelSnapshot.cs:534]: ok great

**Review by jamshidbek7375 — CHANGES_REQUESTED** [review]

File: Web/Ecommerce/Ecommerce/Services/CustomerService.cs [hunk 4]
```diff
@@ -58,13 +45,11 @@ public void Delete(int id)
     }
```
Comment (Ramadan1011) [inline Web/Ecommerce/Ecommerce/Services/CustomerService.cs:46]: remove an empty line

File: Web/Ecommerce/Ecommerce/ViewModels/Customer/LoginViewModel.cs [hunk 5]
```diff
@@ -5,14 +5,17 @@ namespace Ecommerce.ViewModels.Customer
 {
     public class LoginViewModel
     {
-        [Required(ErrorMessage="Email is required")]
-        [MaxLength(50,ErrorMessage="Max 50 characters allowed")]
+        [Required(ErrorMessage = "Email is required")]
+        [MaxLength(50, ErrorMessage = "Max 50 characters allowed")]
         [DisplayName("Email")]
         public string Email { get; set; }
```
Comment (Ramadan1011) [inline Web/Ecommerce/Ecommerce/ViewModels/Customer/LoginViewModel.cs:11]: You may add data type for  email like  [DataType(DataType.EmailAddress)]

File: Web/Ecommerce/Ecommerce/ViewModels/Customer/RegisterViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        [Required(ErrorMessage = "First Name is required")]
+        public required string FirstName { get; set; }
+
+        [Required(ErrorMessage = "Last Name is required")]
+        public required string LastName { get; set; }
+
+
+        [Required(ErrorMessage = "Email is required")]
+        [EmailAddress(ErrorMessage = "Invalid email address")]
+        public required string Email { get; set; }
```
Comment (Ramadan1011) [inline Web/Ecommerce/Ecommerce/ViewModels/Customer/RegisterViewModel.cs:16]: You may add data type for email like [DataType(DataType.EmailAddress)]

**Review by jamshidbek7375 — APPROVED** [review]

**Review by Ramadan1011 — APPROVED** [review]
keep going, bro !

**Pull request merged**